### DISCUSSION TITLE
Back soon to help

### DISCUSSION THREAD

**Discussion by UkooLabs** [SixLabors/ImageSharp.Textures#34] [Announcements]
Looks like as fate would have it im in need of dds encoding primarily for xpr0 decoding. And started to write my own decoders from scratch, realizing I'm being stupid I could continue by adding in any missing code to this project. So expect PRs coming soon. I would like to Private message someone but not sure how to, it's a shame you have not got a discord server yet :). What is currently missing as the highest priority?

**Comment by brianpopow** [comment]
Hi @UkooLabs,

nice to hear from you again! We do not have have a gitter chat channel anymore or discord, but you can find my email in my github profile, if you want to send me private messages.
I would prefer, if we could keep discussions public though, so others can participate and give their opinions.

I tried to continue your work, but it's really a lot to do and for a single person it seems too much to handle.
I mean we already have support for decoding good amount of texture formats and many pixel types and compressions, but there are still alot of things todo.

The main things for me are:
- Basic support of encoding textures. Currently we do not support encoding at all. Maybe in a first step only encoding of uncompressed and only one kind of textures (ktx, maybe)?
- Better unit tests. All the different pixel formats need tests. We need to compare our results to a reference decoder. This will be a painful chore.
 I was thinking of trying out ImageMagick for that, 
as we do in the main ImageSharp project. ImageMagick should support decoding textures i think, but i have not tried that yes. If that does not work out, 
we need to compare our results to reference output, which means we need to provide reference output for all texture format / pixel formats / mipmap's (that's alot).
- The maybe biggest issue I see: I think we need some real life application, which uses the `ImageSharp.Textures` package. We need some feedback, if everything works
 out as expected and is ok to use as it is. I am interested in the field of computer game development, but I am definitely not an expert in that specific field.
 
regards,
Brian

**Comment by JimBobSquarePants** [comment]
I think we need to do a thorough review of the architecture of the library before continuing with a massive focus on cleanup over features.

**Comment by Destructor-Ben** [comment]
I needed DDS encoding and decoding which I was hacking together with my own library, but I found this and thought it would be a better idea to make encoders for this instead. I don't know a whole lot about the DDS format apart from what the [programming guide](https://learn.microsoft.com/en-us/windows/win32/direct3ddds/dx-graphics-dds-pguide) and a dozen other linked articles state (which was enough for me to decode an uncompressed 2D texture without mipmaps), but looking through the decoders, it should be simple enough to reverse what they do.

> I think we need some real life application, which uses the `ImageSharp.Textures` package

I'm making a tool to make Hearts of Iron IV mods that automatically converts images to the correct format, one of which is DDS which is used by the game. This could be a good test DDS encoding and decoding.

Not sure if @UkooLabs still wants to implement this.
